### DISCUSSION TITLE
Testing form actions

### DISCUSSION THREAD

**Discussion by titantwentyone** [filamentphp/filament#9126] [Help]
Couldn't see this in the docs. Given that a livewire component is set up with a custom form and action as follows:

```php
public function getForms(): array
{
    return ['testForm'];
}

public function testForm(Form $form): Form
{
    return $form->schema([
        TextInput::make('whatever')
            ->required(),
        Actions::make([
            Action::make('some_action')
                ->action(/** doing something **/)
    ]);
}
```
With the following test (which fails) I can't find a way to pass the `formName` to `callAction` order to test the action:

```php
livewire(MyPage::class)
    ->fillForm([
        'whatever' => 'something'
    ], 'testForm')
    ->callAction('some_action');
````
This results in `Failed asserting that an action with name [some_action] exists on the [App\Filament\Pages\MyPage] page`. Does such functionality exist?

**Comment by sinnbeck** [comment]
Sadly I dont think its possible to use it like that. It would make sense that you should use `->callAction('testForm.some_action');` but this will break the test as it tries to call the `testForm()` method, without the Form injection. 

Perhaps instead you could add a regular action to the page instead?
https://filamentphp.com/docs/3.x/actions/adding-an-action-to-a-livewire-component

**Comment by kayleegherrmann** [comment]
@titantwentyone Did you ever find a solution to this? I have encountered the same predicament.

**Reply by leandrocfe** [reply]
Could you share your code?

**Reply by kayleegherrmann** [reply]
Very similar to OPs format. I am unable to work out how to write a test for the 'test' action. 

Here is an omitted version
```
public function form(Form $form): Form
    {
        return $form
            ->schema([
                TextInput::make('first_name')
                    ->label('First Name')
                    ->required(),
                Actions::make([
                    Action::make('test')
                        ->label('test)
                        ->action(function () {
                            // Some logic
                            // here that I 
                            // want to test
                            $this->save(); 
                        })
                ]),
            ])->statePath('data')
            ->model($this->user);
    }
```

**Comment by TilsonM17** [comment]
Any solution?
I face the same problem, I need to test my action inside the form, but the above method doesn't work.
I tried another way: https://filamentphp.com/docs/3.x/forms/testing#actions

But the docs are not specific about what **customer_id** is, can anyone help me?

**Reply by leandrocfe** [reply]
Could you share the code please?
